openpype/pipeline/create/creator_plugins.py

```diff
@@ -611,6 +614,52 @@ def get_pre_create_attr_defs(self):
         """
         return self.pre_create_attr_defs
 
+    def get_custom_staging_dir_data(self, instance, task_type=None):
+        """Custom staging directory data.
+
+        Arguments:
+            instance (CreatedInstance): Instance for which we want to get
+                staging dir.
+            task_type (str): Task type. If not set it will be queried from
+        Returns:
+            Tuple[Any, Any]: Tuple of staging dir and is_persistent or None
+        Raises:
+            ValueError - if misconfigured template should be used
+        """
+        from openpype.pipeline import get_custom_staging_dir_info
+
+        task_name = instance["task"]
```

Comment:
Wasn't there a possibility in tray/standalone publisher to publish without a **task** set? And thus `instance["task"]` may not exist?

Reply:
Good point. Need to look into this, since it is really optional in TrayPublisher.

Reply:
fixed